spec/rails_helper.rb

```diff
@@ -664,8 +664,26 @@ def using_session_with_localhost_resolution(name, &block)
       # To work around this problem, we are going to preload all the model schemas before running any system tests so that
       # the lock in ActiveRecord::ModelSchema is not acquired at runtime. This is a temporary workaround while we report
       # the issue to the Rails.
+
+      # TODO: halfvec in AI embedding tables is not supported by Rails
+      # it causes a noisy stderr warning cause it is missing from the PG
+      # typemap.
+      #
+      # This workaround will suppress the noise until Rails adds support
+      suppress_embedding_warnings =
+        proc do |&blk|
+          stderr_orig = $stderr
+          $stderr = StringIO.new
+          blk.call
+          warning = $stderr.string
+          if warning.present? && !warning.include?("failed to recognize type of 'embeddings'")
+            stderr_orig.print warning
+          end
```

Comment:
Perhaps we can remove the string from the warning and print the rest of the warning if present instead of hiding the entire warning?